keras/src/quantizers/gptqutils.py

```diff
@@ -0,0 +1,386 @@
+                        tokenized_doc = np.squeeze(tokenizer(doc["text"]))
+                        if len(tokenized_doc) >= seqlen:
+                            break
+                    except Exception:
+                        docs_for_sampling.remove(doc)
+                        if not docs_for_sampling:
+                            raise ValueError(
+                                "Could not find enough valid documents"
+                                "in the C4 sample."
+                            )
+                        continue
+
+                j = random.randint(0, len(tokenized_doc) - seqlen - 1)
+                sample_slice = tokenized_doc[j : j + seqlen]
+                samples.append(np.reshape(sample_slice, (1, seqlen)))
+
+            return np.array(samples, dtype=np.int32)
+        else:
+            logging.info(f"Warning: No specific alias found for '{dataset}'.")
+            logging.info(
+                f"Attempting to load '{dataset}' directly with its "
+                "default configuration."
+            )
+            d_name = dataset
+            d_config = None  # Use the default configuration for the dataset
+
+        # Default to "text" for wikitext2 and other datasets
+        text_column = "text"
+
+        raw_dataset = load_dataset(d_name, d_config, split="train")
+        text_list = [d[text_column] for d in raw_dataset]
+        full_text = "\n\n".join(text_list)
+        all_tokens = tokenizer.tokenize(full_text)
+
+    else:
+        logging.info("\n==> Using pre-made dataset/generator...")
+        dataset_list = list(dataset)
+
+        if not dataset_list:
+            raise ValueError("Provided dataset is empty.")
+
+        if isinstance(dataset_list[0], str):
+            logging.info("   (Dataset contains strings, tokenizing now...)")
+            full_text = "\n\n".join(dataset_list)
+            all_tokens = tokenizer.tokenize(full_text)
+        else:
+            logging.info("   (Dataset is pre-tokenized, concatenating...)")
+            concatenated_tokens = ops.concatenate(
+                [ops.reshape(s, [-1]) for s in dataset_list], axis=0
+            )
+            all_tokens = ops.convert_to_numpy(concatenated_tokens)
+
+    all_tokens = np.array(all_tokens, dtype=np.int32)
+
+    # --- Step 2: Repeat data if it's too short ---
+    required_tokens = nsamples * seqlen
+    if len(all_tokens) < required_tokens:
+        logging.info(
+            f"Warning: Dataset is too short ({len(all_tokens)} tokens)."
+            " Repeating data to generate {nsamples} samples."
+        )
+        repeats = -(-required_tokens // len(all_tokens))  # Ceiling division
+        all_tokens = np.tile(all_tokens, repeats)
+
+    # --- Step 3: Chunk the token list into samples ---
+    # utils.set_random_seed(seed)
+
+    calibration_samples = []
+    for _ in range(nsamples):
+        # Generate a random starting index
+        start_index = random.randint(0, len(all_tokens) - seqlen - 1)
```

Comment:
use rng instead of random package: https://numpy.org/devdocs/reference/random/generator.html